### PR TITLE
Take input of Sudo Password for CLI Commands

### PR DESCRIPTION
**Pull request by atultherajput**
### :nut_and_bolt: Description: What code changed, and why?
Added sudo_password support for HA cmd in ruby

### :chains: Related Resources
https://chefio.atlassian.net/browse/CHEF-795
https://github.com/chef/automate/pull/7774
### :athletic_shoe: How to Build and Test the Change
- Rebuild `automate-cli` (`rebuild components/automate-cli/`) or can use (`akrishna/automate-cli/0.1.0/20230412082516`), `automate-ha-cluster-ctl` (`rebuild components/automate-cluster-ctl/`) or can use (`akrishna/automate-ha-cluster-ctl/0.1.0/20230411133555`), and `automate-ha-deployment`(`rebuild components/automate-backend-deployment/`) or can use (`akrishna/automate-ha-deployment/0.1.0/20230411141141`).
- Upload the above packages `hab pkg upload <PATH_TO_PACKAGE_HART_FILE>`.
- Create an airgap bundle with new packages `sudo ./chef-automate airgap bundle create -m latest_semver.json --channel dev`. Update this [latest_semver.json](https://packages.chef.io/manifests/dev/automate/latest_semver.json) manifest file with your origins. 
Note: Do not update the `automate-cli` origin in the manifest file,  just install the new `automate-cli` on Bastion.
- Set the environment variable `sudo_password`. For eg. `export sudo_password=1234`.
- Run deployment command with -E flag to preserve environment variable. For eg. `sudo -E ./chef-automate deploy config.toml --airgap-bundle automate-4.5.104.aib`. Refer [here](https://www.petefreitag.com/item/877.cfm) for passing environment variables to the sudo command.
### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
Screenshot 2023-04-12 at 1 42 43 PM

Screenshot 2023-04-12 at 1 41 54 PM

Video: https://progresssoftware.sharepoint.com/:v:/s/ChefCoreC/EacHYiZu7aFBgNkiBpoRPwkB0f_Ngkk_CR2vOxTCdbtMfA?e=janCk5

### CODE REVIEW

**Review by rishabhjhs — APPROVED** [review]

**Review by vipin230 — APPROVED** [review]

**Review by vivekshankar1 — APPROVED** [review]

**Review by punitmundra — APPROVED** [review]
just make sure all pipeline will be green

**Comment by atultherajput** [comment]
Hi @AnkurMundhra could you please review this ticket https://chefio.atlassian.net/browse/CHEF-795 ?

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 1]
```diff
@@ -112,7 +112,6 @@ sudo sed -i 's/SELINUX=enforcing/SELINUX=permissive/g' /etc/selinux/config
    - Give `ssh_user` which has access to all the machines. Example: `ubuntu`
    - Give `ssh_port` in case your AMI is running on custom ssh port, default will be 22.
    - Give the `ssh_key_file` path; this key should have access to all the Machines or VMs.
-   - `sudo_password` is only meant to switch to sudo user. If you have configured a password for the sudo user, please provide it here.
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:115]: You have removed this here, but did not add link to Sudo password page anywhere, how will they know what to do if they have sudo password needed.

Comment (atultherajput) [reply components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:115]: done

**Review by vivek-yadav — APPROVED** [review]

**Pull request merged**